Pi Gateway crashes upon connection (latest dev branch)

Hi,

After installing the latest dev branch on my Pi, I now get the following error when Home Assistant tries to connect (the mys process is then closed).

```
Sep 05 09:06:34 DEBUG New connection from 192.168.1.16
Sep 05 09:06:34 DEBUG GWT:TSA:C=2,CONNECTED
Sep 05 09:06:34 DEBUG GWT:RFC:C=2,MSG=255;255;3;0;20;
Segmentation fault
```

Notice quite a few warnings during build.

```hal/architecture/Linux/drivers/core/GPIO.cpp: In constructor ‘GPIOClass::GPIOClass()’:
hal/architecture/Linux/drivers/core/GPIO.cpp:53:18: warning: ‘%s’ directive writing up to 255 bytes into a region of size 48 [-Wformat-overflow=]
    sprintf(file, "/sys/class/gpio/%s/base", de->d_name);
                  ^~~~~~~~~~~~~~~~~~~~~~~~~
hal/architecture/Linux/drivers/core/GPIO.cpp:53:11: note: ‘sprintf’ output between 22 and 277 bytes into a destination of size 64
    sprintf(file, "/sys/class/gpio/%s/base", de->d_name);
    ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
hal/architecture/Linux/drivers/core/GPIO.cpp:62:18: warning: ‘%s’ directive writing up to 255 bytes into a region of size 48 [-Wformat-overflow=]
    sprintf(file, "/sys/class/gpio/%s/ngpio", de->d_name);
                  ^~~~~~~~~~~~~~~~~~~~~~~~~~
hal/architecture/Linux/drivers/core/GPIO.cpp:62:11: note: ‘sprintf’ output between 23 and 278 bytes into a destination of size 64
    sprintf(file, "/sys/class/gpio/%s/ngpio", de->d_name);
    ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~```
and 
```In file included from ./MySensors.h:431,
                 from examples_linux/mysgw.cpp:82:
In member function ‘char* MyMessage::getString(char*) const’,
    inlined from ‘bool transportSendWrite(uint8_t, MyMessage&)’ at ./core/MyTransport.cpp:987:2,
    inlined from ‘bool transportRouteMessage(MyMessage&)’ at ./core/MyTransport.cpp:560:40,
    inlined from ‘void stUplinkTransition()’ at ./core/MyTransport.cpp:252:29:
./core/MyMessage.cpp:284:17: warning: ‘char* strncpy(char*, const char*, size_t)’ output may be truncated copying between 0 and 25 bytes from a string of length 25 [-Wstringop-truncation]
    (void)strncpy(buffer, this->data, this->getLength());
          ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘char* MyMessage::getString(char*) const’,
    inlined from ‘char* protocolMyMessage2Serial(MyMessage&)’ at ./core/MyProtocol.cpp:88:8:
./core/MyMessage.cpp:284:17: warning: ‘char* strncpy(char*, const char*, size_t)’ output may be truncated copying between 0 and 25 bytes from a string of length 25 [-Wstringop-truncation]
    (void)strncpy(buffer, this->data, this->getLength());
          ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘char* MyMessage::getString(char*) const’,
    inlined from ‘char* protocolMyMessage2Serial(MyMessage&)’ at ./core/MyProtocol.cpp:88:8,
    inlined from ‘bool gatewayTransportSend(MyMessage&)’ at ./core/MyGatewayTransportEthernet.cpp:251:51:
./core/MyMessage.cpp:284:17: warning: ‘char* strncpy(char*, const char*, size_t)’ output may be truncated copying between 0 and 25 bytes from a string of length 25 [-Wstringop-truncation]
    (void)strncpy(buffer, this->data, this->getLength());
          ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./core/MyMessage.cpp: In member function ‘char* MyMessage::getString(char*) const’:
./core/MyMessage.cpp:284:17: warning: ‘char* strncpy(char*, const char*, size_t)’ output may be truncated copying between 0 and 25 bytes from a string of length 25 [-Wstringop-truncation]
    (void)strncpy(buffer, this->data, this->getLength());
          ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In function ‘MyMessage& MyMessage::set(const char*)’,
    inlined from ‘void stParentTransition()’ at ./core/MyTransport.cpp:161:29:
./core/MyMessage.cpp:391:15: warning: ‘char* strncpy(char*, const char*, size_t)’ destination unchanged after copying no bytes [-Wstringop-truncation]
  (void)strncpy(this->data, value, this->getLength());
        ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In function ‘MyMessage& MyMessage::set(const char*)’,
    inlined from ‘void stReadyUpdate()’ at ./core/MyTransport.cpp:312:30:
./core/MyMessage.cpp:391:15: warning: ‘char* strncpy(char*, const char*, size_t)’ destination unchanged after copying no bytes [-Wstringop-truncation]
  (void)strncpy(this->data, value, this->getLength());
        ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In function ‘MyMessage& MyMessage::set(const char*)’,
    inlined from ‘void stIDTransition()’ at ./core/MyTransport.cpp:218:30,
    inlined from ‘void stIDTransition()’ at ./core/MyTransport.cpp:202:6:
./core/MyMessage.cpp:391:15: warning: ‘char* strncpy(char*, const char*, size_t)’ destination unchanged after copying no bytes [-Wstringop-truncation]
  (void)strncpy(this->data, value, this->getLength());
        ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘MyMessage& MyMessage::set(const char*)’,
    inlined from ‘bool present(uint8_t, mysensors_sensor_t, const char*, bool)’ at ./core/MySensorsCore.cpp:379:19,
    inlined from ‘void presentNode()’ at ./core/MySensorsCore.cpp:233:15:
./core/MyMessage.cpp:391:15: warning: ‘char* strncpy(char*, const char*, size_t)’ output truncated before terminating nul copying 10 bytes from a string of the same length [-Wstringop-truncation]
  (void)strncpy(this->data, value, this->getLength());
        ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In function ‘MyMessage& MyMessage::set(const char*)’,
    inlined from ‘void prepareSigningPresentation(MyMessage&, uint8_t)’ at ./core/MySigning.cpp:389:90,
    inlined from ‘bool signerInternalProcessPresentation(MyMessage&)’ at ./core/MySigning.cpp:488:28,
    inlined from ‘bool signerProcessInternal(MyMessage&)’ at ./core/MySigning.cpp:176:42,
    inlined from ‘bool signerProcessInternal(MyMessage&)’ at ./core/MySigning.cpp:171:6:
./core/MyMessage.cpp:391:15: warning: ‘char* strncpy(char*, const char*, size_t)’ destination unchanged after copying no bytes [-Wstringop-truncation]
  (void)strncpy(this->data, value, this->getLength());
        ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In function ‘MyMessage& MyMessage::set(const char*)’,
    inlined from ‘bool request(uint8_t, uint8_t, uint8_t)’ at ./core/MySensorsCore.cpp:429:19:
./core/MyMessage.cpp:391:15: warning: ‘char* strncpy(char*, const char*, size_t)’ destination unchanged after copying no bytes [-Wstringop-truncation]
  (void)strncpy(this->data, value, this->getLength());
        ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In function ‘MyMessage& MyMessage::set(const char*)’,
    inlined from ‘bool requestTime(bool)’ at ./core/MySensorsCore.cpp:434:19:
./core/MyMessage.cpp:391:15: warning: ‘char* strncpy(char*, const char*, size_t)’ destination unchanged after copying no bytes [-Wstringop-truncation]
  (void)strncpy(this->data, value, this->getLength());
        ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘MyMessage& MyMessage::set(const char*)’,
    inlined from ‘bool _processInternalCoreMessage()’ at ./core/MySensorsCore.cpp:482:21:
./core/MyMessage.cpp:391:15: warning: ‘char* strncpy(char*, const char*, size_t)’ output truncated before terminating nul copying 2 bytes from a string of the same length [-Wstringop-truncation]
  (void)strncpy(this->data, value, this->getLength());
        ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In function ‘MyMessage& MyMessage::set(const char*)’,
    inlined from ‘void prepareSigningPresentation(MyMessage&, uint8_t)’ at ./core/MySigning.cpp:389:90,
    inlined from ‘bool signerInternalProcessPresentation(MyMessage&)’ at ./core/MySigning.cpp:488:28,
    inlined from ‘bool signerProcessInternal(MyMessage&)’ at ./core/MySigning.cpp:176:42,
    inlined from ‘bool signerProcessInternal(MyMessage&)’ at ./core/MySigning.cpp:171:6,
    inlined from ‘void transportProcessMessage()’ at ./core/MyTransport.cpp:718:30:
./core/MyMessage.cpp:391:15: warning: ‘char* strncpy(char*, const char*, size_t)’ destination unchanged after copying no bytes [-Wstringop-truncation]
  (void)strncpy(this->data, value, this->getLength());
        ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘MyMessage& MyMessage::set(const char*)’,
    inlined from ‘bool gatewayTransportAvailable()’ at ./core/MyGatewayTransportEthernet.cpp:433:25:
./core/MyMessage.cpp:391:15: warning: ‘char* strncpy(char*, const char*, size_t)’ output truncated before terminating nul copying 25 bytes from a string of the same length [-Wstringop-truncation]
  (void)strncpy(this->data, value, this->getLength());
        ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘MyMessage& MyMessage::set(const char*)’,
    inlined from ‘bool present(uint8_t, mysensors_sensor_t, const char*, bool)’ at ./core/MySensorsCore.cpp:379:19,
    inlined from ‘void presentNode()’ at ./core/MySensorsCore.cpp:233:15,
    inlined from ‘bool gatewayTransportAvailable()’ at ./core/MyGatewayTransportEthernet.cpp:435:16:
./core/MyMessage.cpp:391:15: warning: ‘char* strncpy(char*, const char*, size_t)’ output truncated before terminating nul copying 10 bytes from a string of the same length [-Wstringop-truncation]
  (void)strncpy(this->data, value, this->getLength());
        ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘MyMessage& MyMessage::set(const char*)’,
    inlined from ‘void gatewayTransportProcess()’ at ./core/MyGatewayTransport.cpp:48:26,
    inlined from ‘void gatewayTransportProcess()’ at ./core/MyGatewayTransport.cpp:28:13,
    inlined from ‘void _process()’ at ./core/MySensorsCore.cpp:72:25:
./core/MyMessage.cpp:391:15: warning: ‘char* strncpy(char*, const char*, size_t)’ output truncated before terminating nul copying 10 bytes from a string of the same length [-Wstringop-truncation]
  (void)strncpy(this->data, value, this->getLength());
        ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In function ‘MyMessage& MyMessage::set(const char*)’,
    inlined from ‘void prepareSigningPresentation(MyMessage&, uint8_t)’ at ./core/MySigning.cpp:389:90,
    inlined from ‘void signerPresentation(MyMessage&, uint8_t)’ at ./core/MySigning.cpp:143:28:
./core/MyMessage.cpp:391:15: warning: ‘char* strncpy(char*, const char*, size_t)’ destination unchanged after copying no bytes [-Wstringop-truncation]
  (void)strncpy(this->data, value, this->getLength());
        ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘MyMessage& MyMessage::set(const char*)’,
    inlined from ‘void gatewayTransportProcess()’ at ./core/MyGatewayTransport.cpp:48:26,
    inlined from ‘void gatewayTransportProcess()’ at ./core/MyGatewayTransport.cpp:28:13,
    inlined from ‘void _process()’ at ./core/MySensorsCore.cpp:72:25,
    inlined from ‘void wait(uint32_t)’ at ./core/MySensorsCore.cpp:566:11,
    inlined from ‘void transportReportRoutingTable()’ at ./core/MyTransport.cpp:1077:8:
./core/MyMessage.cpp:391:15: warning: ‘char* strncpy(char*, const char*, size_t)’ output truncated before terminating nul copying 10 bytes from a string of the same length [-Wstringop-truncation]
  (void)strncpy(this->data, value, this->getLength());
        ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~```  

no idea of it's related.

--- a/core/MyProtocol.cpp
+++ b/core/MyProtocol.cpp
@@ -60,7 +60,11 @@ bool protocolSerial2MyMessage(MyMessage &message, char *inputString)
 		}
 	}
 	// payload
-	if (command == C_STREAM) {
+	if (str == NULL) {
+		// no payload, set default value
+		message.set((uint8_t)0);
+	} else if (command == C_STREAM) {
+		// stream payload
 		uint8_t bvalue[MAX_PAYLOAD_SIZE];
 		uint8_t blen = 0;
 		while (*str) {
@@ -72,6 +76,7 @@ bool protocolSerial2MyMessage(MyMessage &message, char *inputString)
 		}
 		message.set(bvalue, blen);
 	} else {
+		// regular payload
 		char *value = str;
 		// Remove trailing carriage return and newline character (if it exists)
 		const uint8_t lastCharacter = strlen(value) - 1;

